go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -627,10 +627,21 @@ func (bnc *BaseNetworkController) deleteNamespaceLocked(ns string) (*namespaceIn
 }
 
 func (bnc *BaseNetworkController) syncNodeManagementPort(node *kapi.Node, switchName, routerName string, hostSubnets []*net.IPNet) ([]net.IP, error) {
-	macAddress, err := util.ParseNodeManagementPortMACAddresses(node, bnc.GetNetworkName())
-	if err != nil {
+	// get mac address from node only for legacy reasons, if it doesn't exist, then calculate it from subnets
+	var macAddress net.HardwareAddr
+	var err error
+	// find suitable MAC address
+	// check node annotation first, to ensure we are not picking a new MAC when one was already configured
+	if macAddress, err = util.ParseNodeManagementPortMACAddresses(node, bnc.GetNetworkName()); err != nil && !util.IsAnnotationNotSetError(err) {
```

Comment:
When configuring the management port in UDN we always use the IP address as base, so there is no point of looking for annotation for UDN networks?
Do you plan to handle UDNs that have the annotation?

Reply:
>When configuring the management port in UDN we always use the IP address as base, so there is no point of looking for annotation for UDN networks?

Right.

>Do you plan to handle UDNs that have the annotation?

Originally I had this same check for if the node had the UDN mac annotation we would use it. This was for handling upgrade where UDNs were already programmed. After thinking about it some, I realized UDN is not GA so we don't need to handle this case. We can make sure this gets backported to 4.17z and pretend UDN never used that annotation :)

Reply:
That shouldn't happen, but I can make a guard checking if this controller is default or not for the annotation check. Does that work for you?

Reply:
I am ok to keep it as is if this is not a possible issue.